### ISSUE TITLE
Incorrect behavior of DowngradeIterablePseudoTypeDeclarationRector

### ISSUE THREAD

**Issue by bobab12** [rectorphp/rector#6662]
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/1ec05862-9f01-60f2-b41a-0bc759881bfb

```php
<?php

final class DemoFile
{
    public function run(): iterable
    {
        return function (): iterable {
            return [];
        };
    }
}
```

### Responsible rules

* `DowngradeIterablePseudoTypeDeclarationRector`

## Expected Behavior

This rector should handle closures as well. Adding`PhpParser\Node\Expr\Closure` to the rector's `getNodeTypes()` works for me.

**Comment by TomasVotruba** [comment]
Thank you for your report and demo link!

Could you send a failing test case in a pull-request, so we have it covered in Rector?
Here is step by step tutorial how to add it: https://github.com/rectorphp/rector/blob/master/docs/how_to_add_test_for_rector_rule.md

**Comment by samsonasik** [comment]
The code actually seems invalid, see https://3v4l.org/Og3ck . The method define `iterable` for return type, but actually returns closure.

**Comment by bobab12** [comment]
> The code actually seems invalid, see https://3v4l.org/Og3ck . The method define `iterable` for return type, but actually returns closure.

It seems you're right, sorry for that. However, removing the return type hint iterable from the function makes the code valid and still shows the rule isn't removing the iterable from the closure.

I'm working on a pull-request as we speak.

**Comment by samsonasik** [comment]
Fixed at https://github.com/rectorphp/rector-src/pull/760

**Comment by bobab12** [comment]
> Fixed at [rectorphp/rector-src#760](https://github.com/rectorphp/rector-src/pull/760)

Thanks!
